ocaml/utils/identifiable.ml

```diff
@@ -226,6 +227,28 @@ module Make_tbl (T : Thing) = struct
 
   let map t f =
     of_map (T_map.map f (to_map t))
+
+  let to_array t =
```

Comment:
This is both better and worse than going via a list:
* Worse: it allocates *more* memory, requiring allocation for both the `Some` and the tuple at every iteration.
* Better: The allocated memory is short-lived, making the allocations almost surely in the minor heap and thus quite cheap.

Instead, it seems possible to do this more directly, using `iter` and an `array option ref`. Yes, that's imperative and unhappy. But we're filling an array here, which basically forces us into this stance.

Reply:
Sigh, too bad we don't have perceus.